crates/asset-importer-rs-post-process/src/steps/flip_uvs.rs

```diff
@@ -1,20 +1,209 @@
+            }
+        }
+    }
+}
+impl std::error::Error for FlipUVsError {}
+
+#[derive(Copy, Clone)]
+pub enum UvFlipVariant {
+    X,
+    Y,
+    Z,
+}
+impl Default for UvFlipVariant {
+    fn default() -> Self {
+        Self::Y
+    }
+}
+
 /// Flip UVs
 #[derive(Default)]
-pub struct FlipUVs;
+pub struct FlipUVs {
+    first_flip: UvFlipVariant,
+    second_flip: Option<UvFlipVariant>,
+}
+impl FlipUVs {
+    fn process_material(&self, material: &mut AiMaterial) -> Result<(), FlipUVsError> {
+        let mut flips = vec![self.first_flip];
+        if let Some(second_flip) = self.second_flip {
+            flips.push(second_flip)
+        }
+        if let Some(prop) = material.get_property_mut(matkey::_AI_MATKEY_UVTRANSFORM_BASE, None, 0)
+        {
+            let transform = bytemuck::from_bytes_mut::<AiUvTransform>(&mut prop.data);
+            for flip in flips.iter() {
+                match flip {
+                    UvFlipVariant::X => {
+                        transform.translation.x *= -1.0;
+                        transform.rotation *= -1.0;
+                    }
+                    UvFlipVariant::Y => {
+                        transform.translation.y *= -1.0;
+                        transform.rotation *= -1.0;
+                    }
+                    _ => {
+                        transform.rotation *= -1.0;
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn process_mesh(&self, mesh: &mut AiMesh) -> Result<(), FlipUVsError> {
+        let mut flips = vec![self.first_flip];
+        if let Some(second_flip) = self.second_flip {
+            flips.push(second_flip)
+        }
+
+        let uv_channel_names = &mesh.texture_coordinate_names;
+
+        let occupied_uv_channels = mesh
+            .texture_coords
+            .iter_mut()
+            .enumerate()
+            .filter_map(|(index, channel_o)| {
+                if channel_o.is_some() {
+                    return Some((index, channel_o.as_mut().unwrap()));
+                } else {
+                    return None;
+                }
+            })
+            .collect::<Vec<_>>();
+
+        for (channel_index, channel) in occupied_uv_channels.into_iter() {
+            for uv in channel.iter_mut() {
+                if uv.z != 0.0 && uv.square_length() > 3.0 {
+                    return Err(FlipUVsError::UvsNotClippedError(
+                        uv_channel_names[channel_index].clone(),
+                        channel_index,
+                    ));
+                }
+                if uv.z == 0.0 && uv.square_length() > 2.0 {
+                    return Err(FlipUVsError::UvsNotClippedError(
+                        uv_channel_names[channel_index].clone(),
+                        channel_index,
+                    ));
+                }
+
+                for flip in flips.iter() {
+                    match flip {
+                        UvFlipVariant::X => uv.x = -uv.x + 1.0f32,
```

Comment:
Minor: AiVector3D is made up of AiReals. There exists a compiler flag to swap between single and double precision, for those strange models where double is needed. There will be a type mistmatch unless you add `as AIReal`.